lib/publiccloud/utils.pm

```diff
@@ -49,6 +49,52 @@ sub select_host_console {
     set_var('TUNNELED', $tunneled) if $tunneled;
 }
 
+sub register_addon {
+    my ($remote, $addon) = @_;
+    my $regcode = get_required_var('SCC_REGCODE');
+    my $arch = get_var('PUBLIC_CLOUD_ARCH') // "x86_64";
+    $arch = "aarch64" if ($arch eq "arm64");
+    record_info($addon, "Going to register '$addon' addon");
+    if ($addon =~ /ltss/) {
+        $regcode = get_required_var('SCC_REGCODE_LTSS');
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), '${VERSION_ID}', $arch, "-r $regcode");
+    } elsif (is_sle('<15') && $addon =~ /tcm|wsm|contm|asmm|pcm/) {
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), '`echo ${VERSION} | cut -d- -f1`', $arch);
+    } elsif (is_sle('<15') && $addon =~ /sdk|we/) {
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), '${VERSION_ID}', $arch);
+    } else {
+        ssh_add_suseconnect_product($remote, get_addon_fullname($addon), undef, $arch);
+    }
+}
+
+sub registercloudguest {
+    my ($my_instance, $remote) = @_;
+    my $regcode = get_required_var('SCC_REGCODE');
+    # not all images currently have registercloudguest pre-installed .
+    # in such a case,we need to regsiter against SCC and install registercloudguest with all needed dependencies and then
+    # unregister and re-register with registercloudguest
+    if ($my_instance->run_ssh_command("which registercloudguest", rc_only => 1, proceed_on_failure => 1) == 1) {
+        $my_instance->retry_ssh_command("sudo SUSEConnect -r $regcode", timeout => 420, retry => 3);
+        register_addon($remote, 'pcm');
+        my $install_packages = ' cloud-regionsrv-client';
+        if (is_azure()) {
+            $install_packages .= ' cloud-regionsrv-client-plugin-azure regionServiceClientConfigAzure regionServiceCertsAzure';
+        }
+        elsif (is_ec2()) {
+            $install_packages .= ' cloud-regionsrv-client-plugin-ec2 regionServiceClientConfigEC2 regionServiceCertsEC2';
+        }
+        elsif (is_gce()) {
+            $install_packages .= ' cloud-regionsrv-client-plugin-gce regionServiceClientConfigGCE regionServiceCertsGCE';
+        }
+        else {
+            die 'Unexpected provider ' . get_var('PUBLIC_CLOUD_PROVIDER');
+        }
+        $my_instance->run_ssh_command("sudo zypper -q -y -n in $install_packages", timeout => 420);
+        $my_instance->run_ssh_command("sudo registercloudguest –clean");
```

Comment:
If this is in the branch where `which registercloudguest` fails, I guess this does not make sense? Or is it implicitly installed in the previous line with one of the other packages?

Reply:
second . see line 79 this package contains `registercloudguest`

Reply:
Ahhh, got it. Mind adding a comment to line 79 that explains that? Something like

```
        my $install_packages = 'cloud-regionsrv-client';          # Includes registercloudguest as well
```

Reply:
done